test/bcreader_tests.jl

```diff
@@ -267,20 +267,29 @@ for FT in (Float32, Float64)
             @test bcf_info.monthly_fields[2] == bcf_info.monthly_fields[1]
             @test bcf_info.segment_length[1] == Int(0)
 
-            #  case 3: date - all_dates[segment_idx + 1] > 2
+            #  case 3: date - all_dates[segment_idx + 1] >= 0
             reset_bcf_info(bcf_info)
-            date = DateTime(bcf_info.all_dates[bcf_info.segment_idx[1] + 1] + Dates.Day(3))
+            bcf_info.segment_idx[1] = bcf_info.segment_idx0[1]
+            date = DateTime(bcf_info.all_dates[bcf_info.segment_idx0[1] + 1] + Dates.Day(3))
```

Comment:
To test the edge case where `date - all_dates[segment_idx + 1] == 0`, maybe we should add a test for case 3 where the `+ Dates.Day(3)` here is removed